server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -169,9 +170,20 @@ public JacksonSmileProvider getJacksonSmileProvider(@Smile ObjectMapper objectMa
 
   static Server makeJettyServer(DruidNode node, ServerConfig config, TLSServerConfig tlsServerConfig)
   {
-    final QueuedThreadPool threadPool = new QueuedThreadPool();
-    threadPool.setMinThreads(config.getNumThreads());
-    threadPool.setMaxThreads(config.getNumThreads());
+    final QueuedThreadPool threadPool;
+    if (config.getQueueSize() == Integer.MAX_VALUE) {
+      threadPool = new QueuedThreadPool();
+      threadPool.setMinThreads(config.getNumThreads());
+      threadPool.setMaxThreads(config.getNumThreads());
+    } else {
+      threadPool = new QueuedThreadPool(
+          config.getNumThreads(),
+          config.getNumThreads(),
+          60000,
```

Comment:
yes, just the default value that is used when you do new QueuedThreadPool() in current code, it isn't/wasn't configurable and i don't think it needs to be.
later socket idle timeout is set which is independent of this and that is indeed configurable.